src/Services/Subscriber/SubscriberUtil.php

```
@@ -152,11 +152,27 @@ public static function verifySubscriber($name, $mobile, $activation, $groupId =
         if ($check_mobile) {
 
-            if ($activation != $check_mobile->activate_key) {
+            // Throttle guesses so the activation code cannot be enumerated. The
+            // counter is per client and number, so a third party sending wrong
+            // codes cannot lock the genuine subscriber out of confirming.
+            $clientIp     = isset($_SERVER['REMOTE_ADDR']) ? sanitize_text_field(wp_unslash($_SERVER['REMOTE_ADDR'])) : '';
+            $attemptKey   = 'wpsms_verify_attempts_' . md5($clientIp . '|' . $mobile);
+            $attemptCount = (int) get_transient($attemptKey);
+
+            if ($attemptCount >= 10) {
+                return new \WP_Error('verify_subscriber', esc_html__('Too many attempts. Please try again later.', 'wp-sms'));
```

PHPStan (2.2.6)

[error] Method WP_SMS\Services\Subscriber\SubscriberUtil::verifySubscriber() should return array|string but returns WP_Error.


(return.type)